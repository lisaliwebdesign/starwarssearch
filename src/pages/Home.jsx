import React from "react";
import { inject, observer } from "mobx-react";
import axios from "axios";
import Autocomplete from "../components/Autocomplete";
import {
  Container, // Wrapper (div) for all the content in this component
  Logo, // The logo image (img)
  Form, // Form wrapping the search
  Label, // Label for the search input
  Loading // Dive for loading content
} from "../elements/home";

import logo from "../images/star-wars-seeklogo.svg";

import PageStore from "../stores/PageStore";

export default class Home extends React.Component {
  state = {
    People: [],
    status: "initial"
  };

  componentDidMount() {
    this.loadData();
  }
  //Get data on page load
  loadData = async () => {
    this.setState({
      status: "searching",
      People: []
    });

    PageStore.loadDefaultData()
      .then(data =>
        this.setState({
          status: "done",
          People: data
        })
      )
      .catch(err => {
        this.setState({
          status: "error"
        });
      });
  };

  render() {
    const { People, status } = this.state;
    return (
      <Container>
        <Logo src={logo} />
        {status === "searching" && (
          <Loading>
            <Label>Loading</Label>
          </Loading>
        )}
        {status === "done" && (
          <Form>
            <Label>Search for Character</Label>
            <Autocomplete
              suggestions={this.state.People}
              serverUrl="https://swapi.co/api/people/?search="
              onSelect={selectedValue => {
                this.props.history.push(`/people/${selectedValue}`);
              }}
            />
          </Form>
        )}
        {status === "error" && (
          <Loading>
            <Label>Oops... error in getting data!</Label>
          </Loading>
        )}
      </Container>
    );
  }
}
