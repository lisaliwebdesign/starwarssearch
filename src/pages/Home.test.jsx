import Home from "./Home";

it("<pages><Home> - renders component", () => {
  const wrapper = render(<Home />);
  expect(wrapper).toMatchSnapshot();
});

it("<pages><Home> - render loading ", () => {
  const wrapper = mount(<Home />);
  const text = wrapper
    .find("label")
    .first()
    .text();
  expect(text).toEqual("Loading");
});

it("<pages><Home> - render error ", () => {
  const wrapper = mount(<Home />);
  wrapper.setState({ status: "error" });
  const text = wrapper
    .find("label")
    .first()
    .text();
  expect(text).toEqual("Oops... error in getting data!");
});

it("<pages><Home> - render with data ", () => {
  const wrapper = mount(<Home />);
  wrapper.setState({ People: ["test1", "test2"], status: "done" });
  const text = wrapper
    .find("label")
    .first()
    .text();
  expect(text).toEqual("Search for Character");
});
