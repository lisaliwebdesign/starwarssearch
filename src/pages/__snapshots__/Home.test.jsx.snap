// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<pages><Home> - renders component 1`] = `
<section
  class="sc-ifAKCX eJQwDj"
>
  <img
    class="sc-gzVnrw hJRzZz"
    src="star-wars-seeklogo.svg"
  >
    
  </img>
</section>
`;
